build.rs

```diff
@@ -14,6 +15,141 @@ use walkdir::WalkDir;
 
 const MLN_GIT_REPO: &str = "https://github.com/maplibre/maplibre-native.git";
 const MLN_REVISION: &str = "8e3899df391b21f9cd1786194cb6fbfa318795da";
+const MLN_RELEASE_TAG: &str = "core-12e0922fc4cadcd88808830e697cfb1d5206c8c9";
+
+#[derive(Debug, Clone, serde::Deserialize)]
+struct ReleaseAsset {
+    name: String,
+    browser_download_url: String,
+    digest: Option<String>,
+}
+
+impl ReleaseAsset {
+    fn hash(&self) -> Option<&str> {
+        self.digest.as_ref()?.strip_prefix("sha256:")
+    }
+}
+
+#[derive(serde::Deserialize)]
+struct Release {
+    assets: Vec<ReleaseAsset>,
+}
+
+/// get release metadata and search for wanted files
+fn fetch_asset_metadata(
```

Comment:
I think it is more rust-y if we move all things strongly relating to `ReleaseAsset` into the impl block instead of like this.